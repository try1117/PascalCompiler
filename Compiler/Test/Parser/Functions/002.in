program test;

function abc(a: integer): integer;
type
  myArr = array[-1..1] of double;
const
  c: myArr = (12, 22, 32);
var
  v: integer = 5 + 10;
begin
end;

procedure proc(var name, surname: string; a: integer = 123; const b: double = 2.0);
var
  prInt: char = 'Y';
  s: string = 'abaca';
begin
end;

begin
end.
