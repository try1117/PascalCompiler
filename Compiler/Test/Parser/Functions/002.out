test : function()
   resultType : Nil

test declarations:
   abc : function(
      a : Integer
   ) resultType : Integer

   abc declarations:
      myArr : Type Array [-1, 1] of Double

      c : Const Array [-1, 1] of Double
      |-- Array
      |        |-- 12.000000
      |        |-- 22.000000
      |        --- 32.000000

      v : Integer
      |-- 15

   proc : function(
      name : Var String
      surname : Var String
      a : Integer
      |-- 123

      b : Const Double
      |-- 2.000000

   ) resultType : Nil

   proc declarations:
      prInt : Char
          |-- Y

      s : String
      |-- abaca

